e2e-test/src/test/java/de/sovity/edc/extension/e2e/db/DataSourceFactory.java

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright (c) 2023 sovity GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       sovity GmbH - initial implementation
+ *
+ */
+
+package de.sovity.edc.extension.e2e.db;
+
+import org.eclipse.edc.spi.persistence.EdcPersistenceException;
+import org.eclipse.edc.sql.datasource.ConnectionFactoryDataSource;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+/**
+ * Create {@link DataSource}s from EDC Config.
+ */
+public class DataSourceFactory {
+
+    /**
+     * Create a new {@link DataSource} from JDBC Credentials.
+     * <br>
+     * This method was extracted into a static method, so we can call it from our Test Code.
+     *
+     * @param jdbcCredentials jdbc credentials
+     * @return {@link DataSource}
+     */
+    public static DataSource fromJdbcCredentials(JdbcCredentials jdbcCredentials) {
+        return new ConnectionFactoryDataSource(() -> newConnection(jdbcCredentials));
```

Comment:
this is problematic. this causes many connections which might cause the postgresql to overrun with connections.

this can only stay like this if it is purely code in tests, and should have a comment on it that warns of the issue that instead of creating a new connection for each required connection, one should use a connection pool with re-use such as HikariCP